packages/peregrine/lib/talons/MagentoRoute/useMagentoRoute.js

```diff
@@ -13,6 +14,13 @@ export const useMagentoRoute = (props = {}) => {
     const { replace } = useHistory();
     const { pathname } = useLocation();
     const [componentMap, setComponentMap] = useRootComponents();
+    const [previousPathname, setPreviousPathname] = useState(null);
+    const [
+        { nextRootComponent },
+        {
+            actions: { setNextRootComponent, setPageLoading }
+        }
+    ] = useAppContext();
```

Comment:
In general, nested destructuring can be tough to follow. Consider a depth of 1 for readability.

```es6
const [appState, appActions] = useAppContext()
const { nextRootComponent } = appState
const { setNextRootComponent, setPageLoading } = appActions
```

Reply:
Fixed!